packages/react/src/jsx.tsx

```diff
@@ -0,0 +1,48 @@
+import { JSXElementConstructor, ReactElement } from 'react'
+
+import { handler } from './events'
+import { ComponentDataState, JSONValue } from './template'
+
+type Props = Record<string, JSONValue | ((args: any) => any)> & {
+  children: JSONValue
+}
+
+// tbd: investigate this
+export const jsxs = jsx
+
+export function jsx(
+  componentFactory: (props: any) => React.ReactElement<Props>,
+  { children, ...passedProps }: Props,
+  key: string
+): ComponentDataState {
+  const { type, props } = componentFactory(passedProps)
+  if (typeof type !== 'string') {
+    throw new Error('Invalid component. Make sure to use server-side version of your components.')
+  }
+  const serialisedProps = Object.fromEntries(
+    Object.entries(props).map(([key, value]) => {
+      if (typeof value === 'function') {
```

Comment:
Here, we automatically wrap functions (e.g. onPress={() => {}}) but we should also automatically wrap objects that are ActionDataState or an array of them.

Reply:
right that makes sense